### PR TITLE
fixed issue 4133 by updating U3 and U2 definitions

### LINKED ISSUE #4133: U3 extension docs list U2
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
https://github.com/Qiskit/qiskit-terra/blob/3b3536bcdb83124d49723dd205573f169c82ea9c/qiskit/extensions/standard/u3.py#L34

### LINKED ISSUE #4395: Simple Latex typo in u3 gate matrix definition
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

Under the u3 gate definition, the 2nd row of the matrix is written (line 48 of u3.py):
                e^{i\phi}\sin(\th) & e^{i(\phi+\lambda)\cos(\th)}
Notice the code(\th) is in the exponent.

### Steps to reproduce the problem

The cos(\th) term should be taken out of the exponent.
### Suggested solutions

It should be written
                e^{i\phi}\sin(\th) & e^{i(\phi+\lambda)}\cos(\th)

### PR DESCRIPTION
**Pull request by FranckChevallier**
### Summary

Fixes #4133 
Fixes #4395

### Details and comments

In U3.py, U3 math definition was actually corresponding to U2 math definition.
In U2.py, U2 matrix representation had a mistake.
In U2.py, U2 math example had a mistake.

This PR fixes both issues.

### CODE REVIEW

File: qiskit/circuit/library/standard_gates/u2.py [hunk 1]
```diff
@@ -42,15 +42,15 @@ class U2Gate(Gate):
 
         U2(\phi, \lambda) = \frac{1}{\sqrt{2}}
             \begin{pmatrix}
-                1          & e^{-i\lambda} \\
+                1          & -e^{-i\lambda} \\
```
Comment (ajavadia) [inline qiskit/circuit/library/standard_gates/u2.py:45]:
```suggestion
                1          & -e^{i\lambda} \\
```

**Comment by ajavadia** [comment]
Also can you update the definition of U3
from:
```
        U3(\theta, \phi, \lambda) =
            \begin{pmatrix}
                \cos(\th)          & e^{-i\lambda}\sin(\th) \\
                e^{i\phi}\sin(\th) & e^{i(\phi+\lambda)\cos(\th)}
            \end{pmatrix}
```
to:
```
        U3(\theta, \phi, \lambda) =
            \begin{pmatrix}
                \cos(\th)          & -e^{i\lambda}\sin(\th) \\
                e^{i\phi}\sin(\th) & e^{i(\phi+\lambda)}\cos(\th)
            \end{pmatrix}
```

two small changes in the bottom right and top right elements.

File: qiskit/circuit/library/standard_gates/u3.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ class U3Gate(Gate):
     Implemented using two X90 pulses on IBM Quantum systems:
 
     .. math::
-        U2(\phi, \lambda) = RZ(\phi+\pi/2) RX(\frac{\pi}{2}) RZ(\lambda-\pi/2)
+        U3(\theta, \phi, \lambda) = RZ(\phi + \pi/2) RX(\theta) RZ(\lambda - \pi/2)
```
Comment (ajavadia) [inline qiskit/circuit/library/standard_gates/u3.py:29]: This also has a mistake. The intention was to write equation 19 here https://arxiv.org/pdf/1612.00858.pdf. So the change should be:
```suggestion
        U3(\theta, \phi, \lambda) = RZ(\phi - \pi/2) RX(\pi/2) RZ(\pi - \theta) RX(\pi/2) RZ(\lambda - \pi/2)
```

**Comment by FranckChevallier** [comment]
Thanks @ajavadia for all your comments.
I've pushed updqted versions of the files.

**Pull request merged**